### PR TITLE
Guide: Protocol Roles

### PR DESCRIPTION
**Pull request by blackgirlbytes**
Link: https://deploy-preview-1538--tbd-website-developer.netlify.app/docs/web5/decentralized-web-nodes/protocol-roles

This pull request includes several updates to improve the documentation and examples related to decentralized web nodes (DWNs) and protocols. The most important changes include updating links to the new protocol documentation, adding a new guide on protocol roles, and adjusting sidebar positions for better organization.

### Documentation Updates:

* [`site/blog/2023-05-23-dwas-vs-pwas.md`](diffhunk://#diff-2cdd8efee4fe535ef7f7fd612ca6e3b1a42db02f4b0c63e7a0d400a54430e4b4L43-R43): Updated the link to the protocols documentation.
* [`site/blog/2023-06-07-did-parental-controls.md`](diffhunk://#diff-db0bc8c5a9daf0ddb10eacf803d61401c8f0727e8737f6d1220693121db5de3fL52-R52): Corrected the link to the protocols documentation.
* [`site/docs/web5/apps/dinger-tutorial.mdx`](diffhunk://#diff-b3a46dbd309c2405e8103804aa2b01e279024a8fd0cb3a892ba459fbbcea55e8L137-R137): Updated multiple links to the new protocols documentation. [[1]](diffhunk://#diff-b3a46dbd309c2405e8103804aa2b01e279024a8fd0cb3a892ba459fbbcea55e8L137-R137) [[2]](diffhunk://#diff-b3a46dbd309c2405e8103804aa2b01e279024a8fd0cb3a892ba459fbbcea55e8L177-R177) [[3]](diffhunk://#diff-b3a46dbd309c2405e8103804aa2b01e279024a8fd0cb3a892ba459fbbcea55e8L206-R206)
* [`site/docs/web5/apps/shared-todo-app.mdx`](diffhunk://#diff-676df17d32913bcc0a3e0dc50a3287532f2daf4463e3ce983d83ad6d7ed8143eL22-R22): Corrected the links to the protocols documentation. [[1]](diffhunk://#diff-676df17d32913bcc0a3e0dc50a3287532f2daf4463e3ce983d83ad6d7ed8143eL22-R22) [[2]](diffhunk://#diff-676df17d32913bcc0a3e0dc50a3287532f2daf4463e3ce983d83ad6d7ed8143eL154-R154)
* [`site/docs/web5/apps/upgrade-to-web5.mdx`](diffhunk://#diff-da0a78cd401a9b4ea40769e44197a01a3de439ca6e2fa087794d6329a615ab47L36-R36): Updated the link to the protocols documentation.
* [`site/docs/web5/decentralized-web-nodes/what-are-dwns.md`](diffhunk://#diff-d35b0c3defb92ed388221a767442c709adbb1ab5f690e344a99a12da06b4ec78L36-R36): Corrected the link to the protocols documentation.
* [`site/docs/web5/decentralized-web-nodes/what-are-protocols.mdx`](diffhunk://#diff-c79c5bc0ddde018d098b7b16879b29ec0b06b1e012030fa7dab2928798ec1347L2-R8): Renamed from `protocols.md` to `what-are-protocols.mdx` and updated the title.
* [`site/netlify.toml`](diffhunk://#diff-2ad05f770dc41cb3585c1729f20869d43ceb8cd29b07bd803392fd8f1dfc19f0L58-R58): Updated the redirect link for protocols documentation.

### New Guide and Examples:

* [`site/docs/web5/decentralized-web-nodes/protocol-roles.mdx`](diffhunk://#diff-2cbc828286538908b0604a451d7b04f848f559284e88fd8455bba7c267e4c0f5R1-R81): Added a new guide on protocol roles, including examples on defining, installing, and using roles.
* [`site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/protocol-roles.test.js`](diffhunk://#diff-67e33ab7512830c0cf1c195cd276cc3f4b3191ca5abfd756f75c8fa92b4196b7R1-R125): Added new test cases for protocol roles, including defining a protocol, installing it, assigning roles, and performing actions within a role.

### Sidebar Adjustments:

* [`site/docs/web5/decentralized-web-nodes/access-data-with-drls.mdx`](diffhunk://#diff-86b0b230381fc7a09b476c26d60e3f942c42ca35d30dba4a9e2925c25ad8bab6L2-R2): Adjusted sidebar position.
* [`site/docs/web5/decentralized-web-nodes/host-dwn-server.mdx`](diffhunk://#diff-ab40248b031a9fd098f1f09f924fc4e06270fdc2a229b30a80d09802eb0c16e1L2-R2): Adjusted sidebar position.
* [`site/docs/web5/decentralized-web-nodes/send-to-dwn.md`](diffhunk://#diff-be5d645a70700b8f5c3434ca8f5443c2f082d236853ef78f94ebb468207290b7L3-R3): Adjusted sidebar position.
* [`site/docs/web5/decentralized-web-nodes/publishing-records.md`](diffhunk://#diff-fc88ace977b4317e557a322c0864f3ec1c381199ba88f9aa0ebc53eacf4069aaL9-R9): Updated the link to the protocols documentation.
* [`site/docs/web5/decentralized-web-nodes/troubleshooting/common-errors.mdx`](diffhunk://#diff-2deb0b70c0f4da1d4ad2b06fa4953780d6b1e89d5fa7c49465ef8885db11c36fL112-R112): Updated the link to the protocols documentation.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205847338399132

### CODE REVIEW

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/protocol-roles.test.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            $role: true,
+        },
+        playlist: {
+            $actions: [ 
+                {
+                    who: 'anyone',
+                    can: ['read']
+                },
+                {
+                    role: 'curator',
```
Comment (EbonyLouis) [inline site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/protocol-roles.test.js:41]: To add more clarity maybe add some highlighting? So highlight the lines mentioned in the paragraph above, not necessary but just a thought.

**Review by EbonyLouis — APPROVED** [review]
Super clear and easy to follow, left one comment

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/protocol-roles.test.js (hunk 1 above)
Comment (blackgirlbytes) [reply site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-web-nodes/protocol-roles.test.js:41]: True! Okay adding the highlighting now

File: site/docs/web5/decentralized-web-nodes/protocol-roles.mdx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+## Built-In Roles
+
+Protocols offer three built-in roles that you can assign permissions to if needed:
+
+* **author**: The entity that created the record.
+* **recipient**: The entity with whom the record is shared.
+* **anyone**: Any user who has access to the application and has installed the protocol.
+
+You can assign actors with these roles the ability to read, write, query, update, delete, or prune data stored on your [Decentralized Web Node (DWN)](/docs/web5/decentralized-web-nodes/what-are-dwns).
+
```
Comment (angiejones) [inline site/docs/web5/decentralized-web-nodes/protocol-roles.mdx:21]: Can you include a code snippet here showing the usage of built-in roles?

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/docs/web5/decentralized-web-nodes/protocol-roles.mdx (hunk 2 above)
Comment (blackgirlbytes) [reply site/docs/web5/decentralized-web-nodes/protocol-roles.mdx:21]: okay done!

**Review by angiejones — APPROVED** [review]

**Pull request merged**